apps/docs/components/SideNavigation.tsx

```diff
@@ -0,0 +1,80 @@
+import useTableContentObserver from "@/hooks/useTableContentObserver";
+import Link from "next/link";
+import { useEffect, useState } from "react";
+
+type Heading = {
+  id: string;
+  text: string | null;
+  level: number;
+};
+
+const SideNavigation = ({ pathname }) => {
+  const [headings, setHeadings] = useState<Heading[]>([]);
+  const [selectedId, setSelectedId] = useState<string | null>(null);
+
+  useTableContentObserver(setSelectedId, pathname);
+
+  useEffect(() => {
+    const getHeadings = () => {
+      const headingElements = document.querySelectorAll("h2[id], h3[id], h4[id]");
+      const headings: Heading[] = Array.from(headingElements).map((heading) => ({
+        id: heading.id,
+        text: heading.textContent,
+        level: parseInt(heading.tagName.slice(1)),
+      }));
+
+      const hasH2 = headings.some((heading) => heading.level === 2);
+
+      setHeadings(hasH2 ? headings : []);
+    };
```

Comment:
Can you please add a few descriptive comments here, so that other contributors can understand the working and the logic behind your approach?

Reply:
Done